app/shared/components/header/header.component.ts

```diff
@@ -1,27 +1,32 @@
-import { Component, Inject } from '@angular/core';
-import { RouterLink, Router } from '@angular/router-deprecated';
+import { Component, OnInit } from '@angular/core';
+import { Router, ROUTER_DIRECTIVES } from '@angular/router';
 import { CollapseDirective, DROPDOWN_DIRECTIVES } from 'ng2-bootstrap';
-import { RootComponent } from '../../../sections/root/root.component';
 import { SearchComponent } from '../search/search.component';
-import { HeaderMenuComponent } from 'ng2-contentful-blog';
+import { HeaderMenuComponent, BreadcrumbsService, BreadcrumbsEvent } from 'ng2-contentful-blog';
 
 import { Angulartics2On } from 'angulartics2/index';
 
 @Component({
   selector: 'gm-header',
   template: require('./header.html') as string,
   styles: [require('./header.styl') as string],
-  directives: [HeaderMenuComponent, CollapseDirective, DROPDOWN_DIRECTIVES, RouterLink, SearchComponent, Angulartics2On]
+  directives: [HeaderMenuComponent, CollapseDirective, DROPDOWN_DIRECTIVES, ROUTER_DIRECTIVES, SearchComponent, Angulartics2On]
 })
-export class HeaderComponent {
+export class HeaderComponent implements OnInit {
   private isOnRootView: boolean;
   private collapsed: boolean = true;
   private router: Router;
+  private breadcrumbsService: BreadcrumbsService;
 
-  public constructor(@Inject(Router) router: Router) {
+  public constructor(router: Router,
+                     breadcrumbsService: BreadcrumbsService) {
     this.router = router;
-    this.router.subscribe((res: any) => {
-      this.isOnRootView = res.instruction.componentType === RootComponent;
+    this.breadcrumbsService = breadcrumbsService;
+  }
+
+  public ngOnInit(): any {
+    this.breadcrumbsService.breadcrumbs$.subscribe((res: BreadcrumbsEvent)=> {
```

Comment:
It would be better to subscribe to router.events and don't use breadcrumbs service - cause it is not his responsibility to tell whether we are on the root page or not.